tests/ignite/handlers/test_checkpoint.py

```diff
@@ -742,7 +752,8 @@ def update_fn(engine, batch):
         assert key in loaded_optimizer_state_dict
         optim_value = optim_state_dict[key]
         loaded_optim_value = loaded_optimizer_state_dict[key]
-        assert optim_value == loaded_optim_value
+        if idist.get_rank() == 0:
```

Comment:
@sdesrozis why do you check for rank == 0 here ?
I have an impression that it works for all ranks ?
cc @sadra-barikbin